web/models/attendance/attendance_helper.ex

```diff
@@ -84,4 +84,110 @@ defmodule CoursePlanner.AttendanceHelper do
 
     Repo.transaction(updated_multi)
   end
+
+  def update_class_attendance_records(offered_course_id) do
+    query = from oc in OfferedCourse,
+    join: s in assoc(oc, :students),
+    join: c in assoc(oc, :classes),
+    join: cs in assoc(c, :students),
```

Comment:
Why do you join class with students? offered course students and class students are the same